activesupport/lib/active_support/duration.rb

```diff
@@ -80,7 +80,7 @@ def ago(time = ::Time.current)
     def inspect #:nodoc:
       parts.
         reduce(::Hash.new(0)) { |h,(l,r)| h[l] += r; h }.
-        sort_by {|unit,  _ | [:years, :months, :days, :minutes, :seconds].index(unit)}.
+        sort_by {|unit,  _ | [:years, :months, :weeks, :days, :hours, :minutes, :seconds].index(unit)}.
```

Comment:
My only concern is that we're going to get different output depending on how the duration is constructed, e.g:

``` irb
>> ActiveSupport::Duration.parse('P1W')
=> 1 week
>> 1.week
=> 7 days
```

Pondering whether we should not decompose values in Rails 5.